streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/SmokeTestDriverIntegrationTest.java

```diff
@@ -140,15 +140,19 @@ public void shouldWorkWithRebalance(
 
 
         final Properties props = new Properties();
+        final String appId = safeUniqueTestName(testInfo);
         props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, safeUniqueTestName(testInfo));
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
         props.put(InternalConfig.STATE_UPDATER_ENABLED, stateUpdaterEnabled);
         props.put(InternalConfig.PROCESSING_THREADS_ENABLED, processingThreadsEnabled);
-        // decrease the session timeout so that we can trigger the rebalance soon after old client left closed
-        props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 10000);
-        props.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 500);
         if (streamsProtocolEnabled) {
             props.put(StreamsConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.STREAMS.name().toLowerCase(Locale.getDefault()));
+            // decrease the session timeout so that we can trigger the rebalance soon after old client left closed
+            CLUSTER.setGroupSessionTimeout(appId, 10000);
+            CLUSTER.setGroupHeartbeatTimeout(appId, 1000);
```

Comment:
The integration test set the session timeout and the heartbeat interval incorrectly before in the new protocol. We need to set it on the group level.

This sometimes made the test flaky with the new assignment algorithm, since we iteratively cycle out the "oldest" member, and tend to assign the tasks from the next-oldest member. But, due to the high session timeout and heartbeat timeout, it could sometimes take too long for the new member to get the new tasks assigned, before being cycled out as well.

I don't see a problem in the assignment logic here - actually, it seems useful to assign tasks to "old" members, since they are stable. We are just cycling out the tasks to quickly in this integration test.